### ISSUE TITLE
uWSGI workers die under gevent 1.5.0

### ISSUE THREAD

**Issue by vytas7** [unbit/uwsgi#2144]
I'm experiencing strange issues with the newly released `gevent==1.5.0`.

Application workers get killed with
```
DAMN ! worker 1 (pid: 26925) died :( trying respawn ...
```
...after a seemingly successful start.

Using `gevent.monkey.patch_all()` helps to alleviate the issue, although previously unseen weird errors are displayed:
```
TypeError: 'module' object is not callable
TypeError: 'module' object is not callable
TypeError: 'module' object is not callable
```

Everything works as expected with `gevent==1.4.0`.

To reproduce, place `case.sh` and `test.py` under `/tmp/test/` in an Ubuntu 16.04 or 18.04 Docker container, and run `case.sh`:
[test.zip](https://github.com/unbit/uwsgi/files/4464288/test.zip)

I'm also getting the same behaviour when building uWSGI Python and Gevent support as plugins, i.e. along the lines of
```
PYTHON=python3.5 make PROFILE=nolang
PYTHON=python3.5 ./uwsgi --build-plugin plugins/python
PYTHON=python3.5 ./uwsgi --build-plugin plugins/gevent
```

**Comment by vytas7** [comment]
FWIW, an application is not even needed to reproduce. Loading `uwsgi` with `gevent==1.5.0` without any app is sufficient to reproduce:
```
uwsgi --http :8000 --gevent 128 --master

<...>

*** Operational MODE: async ***
*** no app loaded. going in full dynamic mode ***
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 5067)
spawned uWSGI worker 1 (pid: 5068, cores: 128)
spawned uWSGI http 1 (pid: 5069)
*** running gevent loop engine [addr:0x484210] ***
DAMN ! worker 1 (pid: 5068) died :( trying respawn ...
Respawned uWSGI worker 1 (new pid: 5070)

...
```

**Comment by anentropic** [comment]
So... what was the fix here?

I'm getting these `TypeError: 'module' object is not callable` with `gevent==20.9.0` and Python 3.8.6

**Comment by vytas7** [comment]
@anentropic IIRC my fix was made available as part of [uWSGI 2.0.19.1](https://pypi.org/project/uWSGI/2.0.19.1/).
My patch addresses the incompatibility with an older version of Gevent than `20.9.0`, so there is no 100% guarantee it would help; but yes, I would obviously recommend giving update a shot.

**Comment by anentropic** [comment]
> @anentropic IIRC my fix was made available as part of [uWSGI 2.0.19.1](https://pypi.org/project/uWSGI/2.0.19.1/).
> My patch addresses the incompatibility with an older version of Gevent than `20.9.0`, so there is no 100% guarantee it would help; but yes, I would obviously recommend giving update a shot.

thanks, I just tried upgrading and it works... `TypeError` is gone 👍 

(sorry, I deleted my previous comment before seeing your reply as I realised I had meant to post it on the linked gevent issue instead, and also I should just try it)
